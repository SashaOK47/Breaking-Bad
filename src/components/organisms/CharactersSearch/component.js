import PropTypes from "prop-types";
import style from "./styles.module.scss";
import Card from "../../molecules/Card";
import LinkBtn from "../../atoms/LinkBtn";

const CharactersSearch = ({
  charactersSearch,
  searchResult,
  clickHandlerCatalog,
  result,
  found,
}) => {
  return (
    <div className={style.charactersSearch}>
      {charactersSearch.length ? (
        <p className={style.result}>
          По вашему запросу <strong>"{searchResult}"</strong> {found}{" "}
          <strong>{charactersSearch.length}</strong> {result}
        </p>
      ) : null}

      {charactersSearch.length ? (
        <div className={style.characters}>
          {charactersSearch.map((character) => (
            <Card character={character} grid={true} key={character.char_id} />
          ))}
        </div>
      ) : (
        <p className={style.noResult}>
          По Вашему запросу <strong>"{searchResult}"</strong> результатов не
          найдено
        </p>
      )}
      <LinkBtn to="#" onClick={clickHandlerCatalog} />
    </div>
  );
};
CharactersSearch.propTypes = {
  charactersSearch: PropTypes.array,
  searchResult: PropTypes.string,
  clickHandlerCatalog: PropTypes.func,
};
export default CharactersSearch;
