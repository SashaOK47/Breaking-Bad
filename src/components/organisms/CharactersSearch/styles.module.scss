@import "../../../assets/styles/main.scss";
.charactersSearch {
  .result,
  .noResult {
    margin-bottom: 2rem;
    font-size: 1.5rem;
    @include tabletLg {
      text-align: center;
    }
  }
  .characters {
    margin-bottom: 2rem;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 1.5rem 2rem;
    @include tabletLg {
      grid-template-columns: repeat(2, 1fr);
    }
    @include custom(675px) {
      grid-template-columns: 1fr;
      justify-items: center;
    }
  }
}
