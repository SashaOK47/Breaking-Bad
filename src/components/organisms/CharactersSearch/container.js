import CharactersSearch from "./component";
import { useSelector } from "react-redux";
import { resetStoreAll } from "../../../store/actions/characters";
import { useDispatch } from "react-redux";

const CharactersSearchContainer = ({ charactersSearch }) => {
  const dispatch = useDispatch();
  const searchQuery = useSelector((state) => state.characters.searchQuery);

  let searchResult;
  searchResult = searchQuery;

  const clickHandlerCatalog = () => {
    dispatch(resetStoreAll());
  };

  let result;
  if (charactersSearch.length === 1) {
    result = "результат";
  } else if (charactersSearch.length >= 2 && charactersSearch.length <= 4) {
    result = "результата";
  } else {
    result = "результатов";
  }

  return (
    <CharactersSearch
      charactersSearch={charactersSearch}
      searchResult={searchResult}
      clickHandlerCatalog={clickHandlerCatalog}
      result={result}
    />
  );
};

export const container = CharactersSearchContainer;
