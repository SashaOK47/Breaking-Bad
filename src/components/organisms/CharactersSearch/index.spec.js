import React from "react";
import CharactersSearch from "./component";

let charactersSearch = [
  {
    char_id: 1,
    name: "Test Test",
    birthday: "12-12-2012",
    img: "test.test",
    status: "Test status",
    nickname: "Test nickname",
  },
  {
    char_id: 2,
    name: "Test Test",
    birthday: "11-11-2011",
    img: "https://test2.test",
    status: "Test2 status",
    nickname: "Test2 nickname",
  },
];

let searchResult = "Test";
const found = "Найдено";

const setUp = (props) => shallow(<CharactersSearch {...props} />);

describe("Character component", () => {
  it("should render CharactersSearch component", () => {
    const component = setUp({ charactersSearch, found, searchResult });
    expect(component).toMatchSnapshot();
  });
  it("should render no result", () => {
    charactersSearch = [];
    searchResult = "123";
    const component = setUp({ charactersSearch, found, searchResult });
    expect(component).toMatchSnapshot();
  });
});
