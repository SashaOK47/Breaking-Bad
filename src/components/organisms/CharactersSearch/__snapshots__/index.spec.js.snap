// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component should render CharactersSearch component 1`] = `
<div
  className="charactersSearch"
>
  <p
    className="result"
  >
    По вашему запросу 
    <strong>
      "
      Test
      "
    </strong>
     
    Найдено
     
    <strong>
      2
    </strong>
     
  </p>
  <div
    className="characters"
  >
    <Card
      character={
        Object {
          "birthday": "12-12-2012",
          "char_id": 1,
          "img": "test.test",
          "name": "Test Test",
          "nickname": "Test nickname",
          "status": "Test status",
        }
      }
      grid={true}
      key="1"
    />
    <Card
      character={
        Object {
          "birthday": "11-11-2011",
          "char_id": 2,
          "img": "https://test2.test",
          "name": "Test Test",
          "nickname": "Test2 nickname",
          "status": "Test2 status",
        }
      }
      grid={true}
      key="2"
    />
  </div>
  <LinkBtn
    to="#"
  />
</div>
`;

exports[`Character component should render no result 1`] = `
<div
  className="charactersSearch"
>
  <p
    className="noResult"
  >
    По Вашему запросу 
    <strong>
      "
      123
      "
    </strong>
     результатов не найдено
  </p>
  <LinkBtn
    to="#"
  />
</div>
`;
